src/main/java/com/faforever/client/teammatchmaking/TeamMatchmakingMapListController.java

```diff
@@ -0,0 +1,198 @@
+  private Pane dialogRoot;
+
+
+  @Override
+  protected void onInitialize() {
+    JavaFxUtil.bindManagedToVisible(loadingPane);
+  }
+
+  @Override
+  public Pane getRoot() {
+    return root;
+  }
+
+
+  public void init(MatchmakerQueueInfo queue, Pane dialogRoot) {
+    this.dialogRoot = dialogRoot;
+    mapService.getMatchmakerBrackets(queue).subscribe(rawBrackets -> {
+      loadingPane.setVisible(false);
+      this.rawBrackets = rawBrackets;
+      this.sortedBracketsWithDuplicates = this.getSortedBrackets(rawBrackets);
+      this.sortedBrackets = this.removeDuplicates(this.getSortedBrackets(rawBrackets));
+
+      PlayerInfo player = playerService.getCurrentPlayer();
+      LeaderboardRating ratingBean = player.getLeaderboardRatings().get(queue.getLeaderboard().technicalName());
+      double rating = ratingBean.mean() - 3 * ratingBean.deviation();
+      this.playerBracketIndex = this.getPlayerBracketIndex(this.sortedBrackets, rating);
+
+      List<MapVersion> values = this.sortedBrackets.values().stream().flatMap(List::stream).toList();
+      this.resizeToContent(values.size(), this.TILE_SIZE);
+
+      Flux.fromIterable(values)
+          .publishOn(fxApplicationThreadExecutor.asScheduler())
+          .doOnNext(this::addMapTile)
+          .subscribe();
+    });
+
+  }
+
+
+  private SortedMap<MatchmakerQueueMapPool, List<MapVersion>> getSortedBrackets(Map<MatchmakerQueueMapPool, List<MapVersion>> brackets) {
+
+    Comparator<MapVersion> mapVersionComparator = Comparator.nullsFirst(Comparator.comparing(MapVersion::size))
+                                                            .thenComparing(mapVersion -> mapVersion.map().displayName(),
+                                                                           String.CASE_INSENSITIVE_ORDER);
+
+    Comparator<MatchmakerQueueMapPool> mapPoolComparator = Comparator.comparingDouble(MatchmakerQueueMapPool::minRating)
+                                                                     .thenComparingDouble(
+                                                                         MatchmakerQueueMapPool::maxRating);
+
+    SortedMap<MatchmakerQueueMapPool, List<MapVersion>> sortedMap = new TreeMap<>(mapPoolComparator);
+
+    brackets.forEach((mapPool, mapVersions) -> {
+      List<MapVersion> sortedList = mapVersions.stream().sorted(mapVersionComparator).collect(Collectors.toList());
+      sortedMap.put(mapPool, sortedList);
+    });
+
+    return sortedMap;
+  }
+
+  private SortedMap<MatchmakerQueueMapPool, List<MapVersion>> removeDuplicates(SortedMap<MatchmakerQueueMapPool, List<MapVersion>> sortedBrackets) {
+    HashSet<String> usedMaps = new HashSet<>();
+    sortedBrackets.replaceAll((k, v) -> sortedBrackets.get(k).stream().filter(mapVersion -> {
+      String name = mapVersion.map().displayName();
+      if (usedMaps.contains(name))
+        return false;
+      usedMaps.add(name);
+      return true;
+    }).collect(Collectors.toList()));
+    return sortedBrackets;
```

Comment:
This is modifying the map in place but the function implies it is returning a new object. So this needs to either return a copy or just modify in place.